### ISSUE TITLE
Cannot query "id" via GraphQL but works in REST

### ISSUE THREAD

**Issue by Matb85** [strapi/strapi#22054]
### Required System information

- Node.js version: v22.9.0
- NPM version: v10.8.3
- Strapi version: v5.2.0
- Database: postgres
- Operating system: Macos 15.0.1
- Is your project Javascript or Typescript: Typescript

### Describe the bug

Cannot query "id" fields via GraphQL (but works in REST)

### Steps to reproduce the behavior

1. Try to query a fila via GraphQL with the"id" field
2. See the error

### Expected behavior

Allow to query the "id" field via GraphQl.

### Screenshots

Error: cannot query the "id" field via GraphQL:

![Screenshot 2024-11-03 at 15 30 23](https://github.com/user-attachments/assets/3e35ea92-cfa0-4411-8ba2-51d80253d9cd)

However, "id" still shows up when using REST:

![Screenshot 2024-11-03 at 15 32 31](https://github.com/user-attachments/assets/54b2e553-2c45-4768-914b-5afe05aa7acd)

**Comment by derrickmehaffy** [comment]
Id is there for some backwards compatibility but it should not be relied on as it will change with even small changes to the entry and for various other reasons.

**Comment by Matb85** [comment]
hmmm, the why is it available in REST?

**Comment by Matb85** [comment]
probably related somewhat: https://github.com/strapi/strapi/issues/22107

**Comment by thienbd203** [comment]
I am unable to update my Image using the documentId at the moment. Could you please guide me on how to resolve this?

**Comment by thienbd203** [comment]
The Lunar New Year is approaching, and I only have 2 days left to resolve this issue. Can anyone help me with this?

**Comment by thienbd203** [comment]
> I am unable to update my Image using the documentId at the moment. GraphQL does not return the id, but it requires the id to perform an update. How can I resolve this issue?

help pls

**Comment by thienbd203** [comment]
Temporary solution
Update config/plugins.ts
graphql: {
    enabled: true,
    config: {
      v4CompatibilityMode: true,
    },
  },

> I am unable to update my Image using the documentId at the moment. GraphQL does not return the id, but it requires the id to perform an update. How can I resolve this issue?

**Comment by bugmaster99** [comment]
> Temporary solution Update config/plugins.ts
> 
> ```
> graphql: {
>     enabled: true,
>     config: {
>       v4CompatibilityMode: true,
>     },
>   },
> ```
> 
> > I am unable to update my Image using the documentId at the moment. GraphQL does not return the id, but it requires the id to perform an update. How can I resolve this issue?

Unfortunately, this is not working. Still getting this issue on frontend when using apollo client v4 queries :-
```
400 Bad Request
"message": "Unknown argument \"id\" on field \"Query.homePage\".",
"code": "GRAPHQL_VALIDATION_FAILED",
"GraphQLError: Unknown argument \"id\" on field \"Query.homePage\".",
```
Below is my `config/plugins.js` for graphql :-
```
graphql: {
      enabled: true,
      config: {
        v4CompatibilityMode: true,
        playgroundAlways: true,
        defaultLimit: 20,
        maxLimit: -1,
        apolloServer: {
          tracing: true,
          introspection: env("ENVIRONMENT_TYPE") === "LOCAL",
        },
      },
    }
```
What am i doing wrong?

**Comment by Matb85** [comment]
> What am i doing wrong?

It is not working for me either because it brings back the old data/attributes nesting @bugmaster99 @thienbd203 .

**Comment by thienbd203** [comment]
"@strapi/plugin-graphql": "^5.8.0",
"@strapi/plugin-users-permissions": "5.8.0",
"@strapi/strapi": "5.8.0",

It still works in this version.

![Image](https://github.com/user-attachments/assets/655a9230-9376-481e-a225-768c7764b51d)

**Comment by quinnlangille** [comment]
Yeah facing the same issue here. I'm a bit confused as to the purpose of the v4Compatibility flag.. it seems I'm still able to fetch data with an ID argument, but it's not defined in the schema so anything with schema validation fails. For example, my already compiled nextjs frontend built using strapi v4 still works when pointed to strapi v5, but I can't run code generation because the "ID" argument is not available in the schema

Can we add the ID argument as deprecated to match the field? I have a large content website that uses the strapi ID in it's URLs, I would like to avoid having to change all my content URLs at the same time as I perform the server update. I'd be happy to make the contribution myself if it's a sound idea

**Comment by derrickmehaffy** [comment]
We currently have no plans to bring back `id` references.

There are a few routes that still use them (some users-permissions and upload/media library routes) but we are entirely moving to string based IDs and both REST and GraphQL will require their usage.

This is largely what we do in major version updates, we issue our breaking changes.
